crates/iota-data-ingestion-core/src/progress_store/file.rs

```diff
@@ -25,7 +107,7 @@ impl ProgressStore for FileProgressStore {
     type Error = IngestionError;
 
     async fn load(&mut self, task_name: String) -> Result<CheckpointSequenceNumber, Self::Error> {
```

Comment:
Is it possible that load and save will be called simultaneously? (I wonder if there is a risk that we will try to write and read the file at the same time)

Reply:
That's a really good question!

Looking at the code flows, concurrent `load` and `save` operations are actually not possible in the current implementation because:
1. Loading happens only during registration of worker pools, before any processing begins.
2. Saving happens during the main processing loop in the `IndexerExecutor`
3. A Single `IndexerExecutor` Task, which is by design